document/concepts/overview.rst

```diff
@@ -1,6 +1,91 @@
+  Each function takes a sequence of values as parameters
+  and returns zero or one values as a result.
+  Functions can call each other, including recursively.
+  The call stack is implicit and cannot be accessed directly.
+  Nor is the address of a function accessible.
+  Functions may also declare mutable *local variables* that are usable as virtual registers.
+
+* **Tables**.
+  A *table* is an array of opaque values of a particular *element type*.
+  It allows programs to select such values indirectly through a dynamic index operand.
+  Currently, the only available element type is an untyped function reference.
+  Thereby, code can perform an indirect call to a function indexed through a table.
+  For example, this allows emulating function pointers with table indices.
+
+* **Linear Memory**.
+  A *linear memory* is a contiguous, mutable array of untyped bytes.
+  Such a memory is created with an initial size but can be dynamically grown.
+  A program can load and store values from/to a linear memory at any byte address (including unaligned) in little endian format.
+  Integer access can specify a *storage size* which is smaller than the size of the respective value type.
+  A trap occurs if access is not within the bounds of the current memory size.
+
+* **Modules**.
+  A WebAssembly binary takes the form of a *module*
+  that contains definitions for functions, tables, and linear memories.
+  It can also define mutable or immutable *global variables*.
+  Each definition can optionally be *exported* under one or more names.
+  Definitions can also be *imported*, specifying a module/name pair and a suitable type.
+
+  To use it, a module must first be *instantiated*.
+  An *instance* is the dynamic representation of a module,
+  complete with its own state and execution stack.
+  Instantiation requires providing definitions for all imports,
+  which may be exports from previously created instances.
+  WebAssembly computations can be initiated by invoking an exported function of an instance.
+  In addition to definitions, a module can define initialization data for its memory or table
+  that takes the form of *segments* copied to given offsets upon instantiation.
+  It can also define a *start function* that is automatically executed after instantiation.
+
+* **Validation**.
+  To be instantiated, a module must be *valid*.
+  Validation performs *type-checking* of functions and their bodies,
+  as well as checking of other well-formedness conditions,
+  e.g., that a module has no duplicate export names.
+
+* **Embedder**.
+  A WebAssembly implementation will typically be *embedded* into a *host* environment.
+  Instantiation of modules and invocation of exports is controlled from this environment.
+  An embedder can also provide external means to create and initialize memories or tables imported by a module.
+  Furthermore, it may allow to supply functions from the host environment as imports to WebAssembly modules.
```

Comment:
The embedder may provide functions from the host environment that can serve as imports to WebAssembly modules.